common/datavalues/src/data_array_arithmetic.rs

```diff
@@ -51,7 +51,8 @@ impl DataArrayArithmetic {
                         scalar.to_array_with_size(1)?
                     }
                 };
-                let result = Self::array_scalar_arithmetic_op(op, &left_array, right_value)?;
+                let right_array = right_value.to_array_with_size(left_array.len())?;
```

Comment:
What's the reason to remove `array_scalar_op` ?

It could have a performance rise with this op.

Reply:
From arrow
```
/// SIMD vectorized version of `divide_scalar`.
#[cfg(feature = "simd")]
fn simd_divide_scalar<T>(
```

Reply:
Aha,I missed `simd_*_scalar`.
From my perf top, the `divide_scalar` function is second performance killer(top 1 is concat_row_to_one_key)
This pull request can be closed now.
Thank you @sundy-li.

Reply:
Ok, I was refactoring the concat_row_to_one_key with vec_hash in #819.

See: https://github.com/datafuselabs/datafuse/blob/f9cbff7a625274c235e355ee70908f9404e3ef48/common/datablocks/src/kernels/data_block_group_by.rs#L76-L87

Reply:
:100: